### ISSUE TITLE
removeFieldBlocs is not working in Flutter

### ISSUE THREAD

**Issue by nilesh2211** [GiancarloCode/form_bloc#276]
**Description**
In Flutter, when i am trying to call removeFieldBlocs function is not removing filed but its adding field i am not sure it is bug or is something missing in my code 

I have checked here : https://giancarlocode.github.io/form_bloc/#/ got to Conditional Field Example this is also not working

**Steps To Reproduce**

class ConditionalFieldsFormBloc extends FormBloc<String, String> {
  final doYouLikeFormBloc = SelectFieldBloc(
    validators: [FieldBlocValidators.required],
    items: ['No', 'Yes'],
  );

  final whyNotYouLikeFormBloc = TextFieldBloc(
    validators: [FieldBlocValidators.required],
  );

  final showSecretField = BooleanFieldBloc();

  final secretField = TextFieldBloc(
    validators: [FieldBlocValidators.required],
  );

  ConditionalFieldsFormBloc() {
    addFieldBlocs(
      fieldBlocs: [
        doYouLikeFormBloc,
      ],
    );

    showSecretField.onValueChanges(
      onData: (previous, current) async* {
        if (current.value) {
          addFieldBlocs(fieldBlocs: [secretField]);
        } else {
          removeFieldBlocs(fieldBlocs: [secretField]);
        }
      },
    );

    doYouLikeFormBloc.onValueChanges(
      onData: (previous, current) async* {
        removeFieldBlocs(
          fieldBlocs: [
            whyNotYouLikeFormBloc,
            showSecretField,
            secretField,
          ],
        );

        if (current.value == 'No') {
          addFieldBlocs(fieldBlocs: [
            whyNotYouLikeFormBloc,
          ]);
        } else if (current.value == 'Yes') {
          addFieldBlocs(fieldBlocs: [
            showSecretField,
            if (showSecretField.value) secretField,
          ]);
        }
      },
    );
  }

  @override
  Future<void> close() {
    whyNotYouLikeFormBloc.close();
    showSecretField.close();
    secretField.close();

    return super.close();
  }
}

**Comment by BreX900** [comment]
Of course this is a bug. It will be fixed in the next release

**Comment by ben700** [comment]
I'm also waiting for fix

**Comment by Ezrabeer** [comment]
Any updates on this?

**Comment by ryuuzake** [comment]
My fix for this problem now is to use the `removeFieldBloc` method. Which surprisingly works well.
So you need to remove the `fieldBloc` individually.

Example
```
removeFieldBloc(fieldBloc: whyNotYouLikeFormBloc);
removeFieldBloc(fieldBloc: showSecretField);
removeFieldBloc(fieldBloc: secretField);
```

**Comment by AHHasanian** [comment]
I have the same problem, it adds instead of deleting.
When will this problem be solved?

**Comment by SimoneBressan** [comment]
@WahdanZ Can you release a new version? This bug is already fixed in master

**Comment by WahdanZ** [comment]
okay i'll sorry for the delay but the upcoming days i'll focus on maintaining the package

**Comment by WahdanZ** [comment]
fix on version 0.30.0
